fix issue 114

jsnyders opened the pull request:
I believe this is one way to fix the issue 114. When a <% %> template was used as a dictionary value the %'s were not being stripped. 
I tested the change with this group template
```
delimiters "$","$"
d1 ::= [
  "p": <%hi%>
]
main() ::= <<$d1.p$>>
```
and the problem was fixed (expected result `hi` was returned).

Fixes #114

sharwell commented:
I think you have the right idea for a solution. Do you mind adding a test to help prevent future regressions? My suggestion would be to place it [here](https://github.com/antlr/stringtemplate4/blob/ce9f65b755b2779b7f018e3e46f50f03b04dc3f6/test/org/stringtemplate/v4/test/TestDictionaries.java#L494), and include a [comment like this](https://github.com/antlr/stringtemplate4/blob/ce1a41b6d8d9987905e72195d0d9a4c84b53a808/test/org/stringtemplate/v4/test/TestSTRawGroupDir.java#L102-L107) so everyone knows it was added for a specific reason. If you have any questions, or if you don't have time and wish for one of the other authors to create the test instead, please let us know. :smile:

sharwell commented:
:exclamation: You'll need to add your name to [**contributors.txt**](https://github.com/antlr/stringtemplate4/blob/master/contributors.txt) before we can accept code contributions. Note that this is a one-time thing. If you prefer, you can add your name to the [ANTLR 4 contributors list instead](https://github.com/antlr/antlr4/blob/master/contributors.txt), which will cover StringTemplate as well as other projects in the ANTLR organization.

jsnyders commented:
I will add a test case. My Java skills are rusty. I have to remember how to run jUnit tests.
I will add my name to contributors.txt

jsnyders commented:
I added a test case and updated contributors.txt in the same commit. Let me know if I didn't do it correctly.
It would be helpful if the ant script had a target for building and running the tests but I figured it out.

parrt commented:
thanks! Sorry it took so long ;)
